presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestMergeTDigestFunction.java

```diff
@@ -0,0 +1,134 @@
+
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.DoubleType;
+import com.facebook.presto.spi.type.SqlVarbinary;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeParameter;
+import com.facebook.presto.tdigest.TDigest;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import static com.facebook.presto.operator.aggregation.AggregationTestUtils.assertAggregation;
+import static com.facebook.presto.spi.type.TDigestParametricType.TDIGEST;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static java.util.Objects.requireNonNull;
+
+public class TestMergeTDigestFunction
+        extends AbstractTestAggregationFunction
+{
+    public static final BiFunction<Object, Object, Boolean> TDIGEST_EQUALITY = (actualBinary, expectedBinary) -> {
+        if (actualBinary == null && expectedBinary == null) {
+            return true;
+        }
+        requireNonNull(actualBinary, "actual value was null");
+        requireNonNull(expectedBinary, "expected value was null");
+
+        TDigest actual = new TDigest(wrappedBuffer(((SqlVarbinary) actualBinary).getBytes()));
+        TDigest expected = new TDigest(wrappedBuffer(((SqlVarbinary) expectedBinary).getBytes()));
+        return actual.getSize() == expected.getSize() &&
+                actual.getMin() == expected.getMin() &&
+                actual.getMax() == expected.getMax() &&
+                actual.getCompressionFactor() == expected.getCompressionFactor();
+    };
+
+    @Override
+    public Block[] getSequenceBlocks(int start, int length)
+    {
+        Type type = TDIGEST.createType(typeRegistry, ImmutableList.of(TypeParameter.of(DoubleType.DOUBLE)));
+        BlockBuilder blockBuilder = type.createBlockBuilder(null, length);
+        for (int i = start; i < start + length; i++) {
+            TDigest tdigest = new TDigest(100);
+            tdigest.add(i);
+            type.writeSlice(blockBuilder, tdigest.serialize());
+        }
+        return new Block[] {blockBuilder.build()};
+    }
+
+    @Override
+    protected String getFunctionName()
+    {
+        return "merge";
+    }
+
+    @Override
+    protected List<String> getFunctionParameterTypes()
+    {
+        return ImmutableList.of("tdigest(double)");
+    }
+
+    @Override
+    public Object getExpectedValue(int start, int length)
+    {
+        if (length == 0) {
+            return null;
+        }
+
+        TDigest tdigest = new TDigest(100);
+        for (int i = start; i < start + length; i++) {
+            tdigest.add(i);
+        }
+        return new SqlVarbinary(tdigest.serialize().getBytes());
+    }
+
+    // The following tests are overridden because by default simple equality checks are done, which often won't work with
```

Comment:
This comment looks familiar.  I think it's an established convention now and we don't need to call it out like I did before.

Reply:
Is this comment resolved?